prow/config/tide_test.go

```diff
@@ -43,3 +45,34 @@ func TestTideQuery(t *testing.T) {
 	checkTok("-label:\"foo\"")
 	checkTok("review:approved")
 }
+
+func TestMergeMethod(t *testing.T) {
```

Comment:
Using an initialized `Tide{}` here with data would allow you to not involve YAML parsing + unmarshalling in your test.

Reply:
The reason I put the yaml into the test was for readability. I was looking through some of the other tests and found it hard to understand what the yaml would look like mapped mapped to how the functionality worked. By putting the yaml in the test the structure is documented by the test.

This was my first code contribution to test-infra and I was looking to make things a bit easier for the next person who came along to understand the structure.

If you want me to change it I can.

Reply:
The code you wrote -- to determine a merge method for a repo -- is what I would expect to be under test in this scenario. YAML unmarshalling is out of scope for the unit test. There is a test that loads the configuration file in checked into the repo, that test is perhaps what you were looking for?

Reply:
Updated